config/connection_settings_test.go

```diff
@@ -451,9 +489,28 @@ func mockProvisioningService(t *testing.T, controller *gomock.Controller, device
 	return provisioningService
 }
 
+func mockProvisioningServiceWithIDScope(t *testing.T, controller *gomock.Controller, deviceData *config.AzureDeviceData, deviceDataError error, hasProvisioningFile bool, timesGetDataCalled, timesInitCalled int) *mock.MockProvisioningService {
+	provisioningService := mock.NewMockProvisioningService(controller)
+	provisioningService.EXPECT().GetDeviceData("dummyIdScope", gomock.Any()).Return(deviceData, deviceDataError).Times(timesGetDataCalled)
+	provisioningService.EXPECT().Init(gomock.Any(), gomock.Any()).Do(func(client config.ProvisioningHTTPClient, provisioningFile io.ReadWriter) {
+		if hasProvisioningFile != (client == nil) {
+			t.Fail()
```

Comment:
Here, this fail can happen in different combination i.e. with or without provisioning file. I would suggest to add a fail reason message.

Reply:
This method is a copy of another one from the initial contribution source code, I suppose there are other places that would need fail reason message and I would suggest to open another ticket for fixing such issues, not on just one place.

Reply:
@stoyan-zoubev - any news here? Did you have time to check for similar issues? I cannot see new related issues in the project. In any case, I would suggest to fix what's found in the current PR.